src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -616,6 +617,9 @@ private void tree(AnalysisSubmission submission, Model model) throws IOException
 
 		Analysis analysis = submission.getAnalysis();
 		AnalysisOutputFile file = analysis.getAnalysisOutputFile(treeFileKey);
+		if (file == null) {
```

Comment:
This may not be the cleanest way to check if a tree was generated. It may be better to insert a check before the `tree()` method is called.

Reply:
This solutions works great. Thanks 😄.  I think it could be simplified further by adding an additional case to the `catch` statement in the code that calls this method https://github.com/dfornika/irida/blob/de56393ddd86b1da2f235ac443300c7e22b843cb/src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java#L231.  That is:

```java
catch (IOException | NullPointerException e) {
  logger.error("Couldn't get preview for analysis", e);
}
```

That way you can remove throwing an additional exception here and just handling the original `NullPointerException`.  And this code will now also properly work for the other preview types.

Reply:
Looking over this code again, I think it's probably best to leave it as it is.  Adding `NullPointerException` to the `catch` statement will catch all `NullPointerException`s, even those unrelated to a missing tree.  We may want those to trigger a user-visible error instead of just hiding the tree.

So, you can leave this code as is.